internal/jimm/bootstrap/bootstrap_test.go

```diff
@@ -0,0 +1,137 @@
+// Copyright 2025 Canonical.
+
+package bootstrap_test
+
+import (
+	"context"
+	"fmt"
+	"math"
+	"testing"
+	"time"
+
+	qt "github.com/frankban/quicktest"
+	"github.com/frankban/quicktest/qtsuite"
+	"github.com/google/uuid"
+
+	"github.com/canonical/jimm/v3/internal/db"
+	"github.com/canonical/jimm/v3/internal/dbmodel"
+	"github.com/canonical/jimm/v3/internal/jimm/bootstrap"
+	"github.com/canonical/jimm/v3/internal/jobtracker"
+	"github.com/canonical/jimm/v3/internal/openfga"
+	"github.com/canonical/jimm/v3/internal/testutils/jimmtest"
+	"github.com/canonical/jimm/v3/pkg/api/params"
+)
+
+type bootstrapManagerSuite struct {
+	manager    *bootstrap.BootstrapManager
+	jobTracker *jobtracker.Tracker
+	adminUser  *openfga.User
+	db         *db.Database
+	ofgaClient *openfga.OFGAClient
+}
+
+func (s *bootstrapManagerSuite) Init(c *qt.C) {
+	db := &db.Database{
+		DB: jimmtest.PostgresDB(c, time.Now),
+	}
+	err := db.Migrate(context.Background())
+	c.Assert(err, qt.IsNil)
+
+	s.db = db
+
+	ofgaClient, _, _, err := jimmtest.SetupTestOFGAClient(c.Name())
+	c.Assert(err, qt.IsNil)
+
+	s.ofgaClient = ofgaClient
+
+	jobtracker, err := jobtracker.NewJobTracker(db, 1*time.Minute)
+	s.jobTracker = jobtracker
+	c.Assert(err, qt.IsNil)
+	s.manager, err = bootstrap.NewBootstrapManager(db, ofgaClient, jobtracker)
+	c.Assert(err, qt.IsNil)
+}
+
+func (s *bootstrapManagerSuite) TestGetBootstrapStatusAndLogs(c *qt.C) {
+	ctx := c.Context()
+	read := make(chan struct{})
+	defer close(read)
+	write := make(chan struct{})
+	defer close(write)
+	numLogs := 101
+	batchSize := 10
```

Comment:
I think a simpler test where we just read 1, then 2, then 1 again would read better than batching.